dataprofiler/profilers/profile_builder.py

```diff
@@ -639,57 +666,108 @@ def tqdm(l):
+                          "not the whole dataset.".format(sample_size))
+        if pool is not None:
+
+            # Create a bunch of simultaneous column conversions
+            for col in df.columns:
+                if min_true_samples is None:
+                    min_true_samples = profile[col]._min_true_samples
+                try:
+                    multi_process_dict[col] = pool.apply_async(
+                        profile[col].clean_data_and_get_base_stats,
+                        (df[col], sample_size, min_true_samples, sample_ids))
+                except Exception as e:
+                    print(e)
+                    single_process_list.add(col)
+                    
+            # Iterate through multiprocessed columns collecting results
+            print(notification_str)
+            for col in tqdm(multi_process_dict.keys()):
+                try:
+                    clean_sampled_dict[col], base_stats = \
+                        multi_process_dict[col].get()
+                    profile[col]._update_base_stats(base_stats)
+                except Exception as e:
+                    print(e)
+                    single_process_list.add(col)
+
+            # Clean up any columns which errored
+            if len(single_process_list) > 0:
+                print("Errors in multiprocessing occured:",
+                      len(single_process_list), "errors, reprocessing...")
+                for col in tqdm(single_process_list):
+                    if min_true_samples is None:
+                        min_true_samples = profile[col]._min_true_samples
+                    clean_sampled_dict[col], base_stats = \
+                        profile[col].clean_data_and_get_base_stats(
+                            df[col], sample_size, min_true_samples, sample_ids)
+                    profile[col]._update_base_stats(base_stats)
+            
+            pool.close() # Close pool for new tasks
+            pool.join() # Wait for all workers to complete
 
+        else: # No pool
+
+            print(notification_str)
+            for col in tqdm(df.columns):
+                if min_true_samples is None:
+                    min_true_samples = profile[col]._min_true_samples
+                clean_sampled_dict[col], base_stats = \
+                    profile[col].clean_data_and_get_base_stats(
+                        df_series=df[col], sample_size=sample_size,
+                        min_true_samples=min_true_samples, sample_ids=sample_ids)
+                profile[col]._update_base_stats(base_stats)
+            
+        # Process and label the data
+        notification_str = "Calculating the statistics... "
+        pool = None
+        if options.structured_options.multiprocess.is_enabled:
+            pool, pool_size = utils.generate_pool(max_pool_size=4)
+            if pool:
+                notification_str += " (with " + str(pool_size) + " processes)"
```

Comment:
test all the notifications are working?

Reply:
There's not a good way to do that in this case with multiple different notification types (tqdm, print) and multiple sets of text.

Further, this isn't critical and we weren't testing it before?

Reply:
It's a non-critical component that has never been tested. A bug here is not likely nor important. I don't see a reason to test it and add more time to our 15 min of tests